src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -28,24 +29,23 @@ public Task RunAsync(TagHelperExecutionContext executionContext)
             }
 
             var tagHelperContext = executionContext.Context;
-            var tagHelpers = executionContext.TagHelpers;
-            OrderTagHelpers(tagHelpers);
+            var tagHelpers = CollectionsMarshal.AsSpan(executionContext.TagHelperList);
 
-            // Read interface .Count once rather than per iteration
-            var count = tagHelpers.Count;
-            for (var i = 0; i < count; i++)
+            tagHelpers.Sort(default(SortTagHelpers));
+
+            foreach (var tagHelper in tagHelpers)
```

Comment:
Why iterate over the sequence two separate ways?

Reply:
Want `i` in the second one?